### PR TITLE
New Source: KIng's Lynn & West Norfolk, UK

### PR DESCRIPTION
**Pull request by dt215git**
New source, as requested in #1088.
This returns ~1 month of collections, not the 6 month calendar. A future update can deal with that longer term schedule.
<details><summary>Tests</summary>

```bash
Testing source west_norfolk_gov_uk ...
  found 6 entries for Test_001
    2023-07-20 : Black sack [mdi:trash-can]
    2023-07-27 : Green sack [mdi:recycle]
    2023-07-27 : Black sack [mdi:trash-can]
    2023-08-03 : Black sack [mdi:trash-can]
    2023-08-10 : Green sack [mdi:recycle]
    2023-08-10 : Black sack [mdi:trash-can]
  found 7 entries for Test_002
    2023-07-19 : Black bin [mdi:trash-can]
    2023-07-26 : Green bin [mdi:recycle]
    2023-07-27 : Brown bin [mdi:leaf]
    2023-08-02 : Black bin [mdi:trash-can]
    2023-08-09 : Green bin [mdi:recycle]
    2023-08-10 : Brown bin [mdi:leaf]
    2023-08-16 : Black bin [mdi:trash-can]
  found 8 entries for Test_003
    2023-07-17 : Brown bin [mdi:leaf]
    2023-07-17 : Black bin [mdi:trash-can]
    2023-07-24 : Green bin [mdi:recycle]
    2023-07-31 : Brown bin [mdi:leaf]
    2023-07-31 : Black bin [mdi:trash-can]
    2023-08-07 : Green bin [mdi:recycle]
    2023-08-14 : Brown bin [mdi:leaf]
    2023-08-14 : Black bin [mdi:trash-can]
  found 6 entries for Test_004
    2023-07-20 : Black sack [mdi:trash-can]
    2023-07-27 : Green sack [mdi:recycle]
    2023-07-27 : Black sack [mdi:trash-can]
    2023-08-03 : Black sack [mdi:trash-can]
    2023-08-10 : Green sack [mdi:recycle]
    2023-08-10 : Black sack [mdi:trash-can]
```

</details>

### CODE REVIEW

**Comment by dt215git** [comment]
Now works for extended collection schedule.
```bash
Testing source west_norfolk_gov_uk ...
  found 39 entries for Test_001
    2023-07-06 : refuse [mdi:trash-can]
    2023-07-13 : recycling [mdi:recycle]
    2023-07-13 : refuse [mdi:trash-can]
    2023-07-20 : refuse [mdi:trash-can]
    2023-07-27 : recycling [mdi:recycle]
    2023-07-27 : refuse [mdi:trash-can]
    2023-08-03 : refuse [mdi:trash-can]
    2023-08-10 : recycling [mdi:recycle]
    2023-08-10 : refuse [mdi:trash-can]
    2023-08-17 : refuse [mdi:trash-can]
    2023-08-24 : recycling [mdi:recycle]
    2023-08-24 : refuse [mdi:trash-can]
    2023-08-31 : refuse [mdi:trash-can]
    2023-09-07 : recycling [mdi:recycle]
    2023-09-07 : refuse [mdi:trash-can]
    2023-09-14 : refuse [mdi:trash-can]
    2023-09-21 : recycling [mdi:recycle]
    2023-09-21 : refuse [mdi:trash-can]
    2023-09-28 : refuse [mdi:trash-can]
    2023-10-05 : recycling [mdi:recycle]
    2023-10-05 : refuse [mdi:trash-can]
    2023-10-12 : refuse [mdi:trash-can]
    2023-10-19 : recycling [mdi:recycle]
    2023-10-19 : refuse [mdi:trash-can]
    2023-10-26 : refuse [mdi:trash-can]
    2023-11-02 : recycling [mdi:recycle]
    2023-11-02 : refuse [mdi:trash-can]
    2023-11-09 : refuse [mdi:trash-can]
    2023-11-16 : recycling [mdi:recycle]
    2023-11-16 : refuse [mdi:trash-can]
    2023-11-23 : refuse [mdi:trash-can]
    2023-11-30 : recycling [mdi:recycle]
    2023-11-30 : refuse [mdi:trash-can]
    2023-12-07 : refuse [mdi:trash-can]
    2023-12-14 : recycling [mdi:recycle]
    2023-12-14 : refuse [mdi:trash-can]
    2023-12-21 : refuse [mdi:trash-can]
    2023-12-28 : recycling [mdi:recycle]
    2023-12-28 : refuse [mdi:trash-can]
  found 39 entries for Test_002
    2023-07-05 : refuse [mdi:trash-can]
    2023-07-12 : recycling [mdi:recycle]
    2023-07-13 : garden [mdi:leaf]
    2023-07-19 : refuse [mdi:trash-can]
    2023-07-26 : recycling [mdi:recycle]
    2023-07-27 : garden [mdi:leaf]
    2023-08-02 : refuse [mdi:trash-can]
    2023-08-09 : recycling [mdi:recycle]
    2023-08-10 : garden [mdi:leaf]
    2023-08-16 : refuse [mdi:trash-can]
    2023-08-23 : recycling [mdi:recycle]
    2023-08-24 : garden [mdi:leaf]
    2023-08-30 : refuse [mdi:trash-can]
    2023-09-06 : recycling [mdi:recycle]
    2023-09-07 : garden [mdi:leaf]
    2023-09-13 : refuse [mdi:trash-can]
    2023-09-20 : recycling [mdi:recycle]
    2023-09-21 : garden [mdi:leaf]
    2023-09-27 : refuse [mdi:trash-can]
    2023-10-04 : recycling [mdi:recycle]
    2023-10-05 : garden [mdi:leaf]
    2023-10-11 : refuse [mdi:trash-can]
    2023-10-18 : recycling [mdi:recycle]
    2023-10-19 : garden [mdi:leaf]
    2023-10-25 : refuse [mdi:trash-can]
    2023-11-01 : recycling [mdi:recycle]
    2023-11-02 : garden [mdi:leaf]
    2023-11-08 : refuse [mdi:trash-can]
    2023-11-15 : recycling [mdi:recycle]
    2023-11-16 : garden [mdi:leaf]
    2023-11-22 : refuse [mdi:trash-can]
    2023-11-29 : recycling [mdi:recycle]
    2023-11-30 : garden [mdi:leaf]
    2023-12-06 : refuse [mdi:trash-can]
    2023-12-13 : recycling [mdi:recycle]
    2023-12-14 : garden [mdi:leaf]
    2023-12-20 : refuse [mdi:trash-can]
    2023-12-27 : recycling [mdi:recycle]
    2023-12-28 : garden [mdi:leaf]
  found 39 entries for Test_003
    2023-07-03 : garden [mdi:leaf]
    2023-07-03 : refuse [mdi:trash-can]
    2023-07-10 : recycling [mdi:recycle]
    2023-07-17 : garden [mdi:leaf]
    2023-07-17 : refuse [mdi:trash-can]
    2023-07-24 : recycling [mdi:recycle]
    2023-07-31 : garden [mdi:leaf]
    2023-07-31 : refuse [mdi:trash-can]
    2023-08-07 : recycling [mdi:recycle]
    2023-08-14 : garden [mdi:leaf]
    2023-08-14 : refuse [mdi:trash-can]
    2023-08-21 : recycling [mdi:recycle]
    2023-08-28 : garden [mdi:leaf]
    2023-08-28 : refuse [mdi:trash-can]
    2023-09-04 : recycling [mdi:recycle]
    2023-09-11 : garden [mdi:leaf]
    2023-09-11 : refuse [mdi:trash-can]
    2023-09-18 : recycling [mdi:recycle]
    2023-09-25 : garden [mdi:leaf]
    2023-09-25 : refuse [mdi:trash-can]
    2023-10-02 : recycling [mdi:recycle]
    2023-10-09 : garden [mdi:leaf]
    2023-10-09 : refuse [mdi:trash-can]
    2023-10-16 : recycling [mdi:recycle]
    2023-10-23 : garden [mdi:leaf]
    2023-10-23 : refuse [mdi:trash-can]
    2023-10-30 : recycling [mdi:recycle]
    2023-11-06 : garden [mdi:leaf]
    2023-11-06 : refuse [mdi:trash-can]
    2023-11-13 : recycling [mdi:recycle]
    2023-11-20 : garden [mdi:leaf]
    2023-11-20 : refuse [mdi:trash-can]
    2023-11-27 : recycling [mdi:recycle]
    2023-12-04 : garden [mdi:leaf]
    2023-12-04 : refuse [mdi:trash-can]
    2023-12-11 : recycling [mdi:recycle]
    2023-12-18 : garden [mdi:leaf]
    2023-12-18 : refuse [mdi:trash-can]
    2023-12-25 : recycling [mdi:recycle]
  found 39 entries for Test_004
    2023-07-06 : refuse [mdi:trash-can]
    2023-07-13 : recycling [mdi:recycle]
    2023-07-13 : refuse [mdi:trash-can]
    2023-07-20 : refuse [mdi:trash-can]
    2023-07-27 : recycling [mdi:recycle]
    2023-07-27 : refuse [mdi:trash-can]
    2023-08-03 : refuse [mdi:trash-can]
    2023-08-10 : recycling [mdi:recycle]
    2023-08-10 : refuse [mdi:trash-can]
    2023-08-17 : refuse [mdi:trash-can]
    2023-08-24 : recycling [mdi:recycle]
    2023-08-24 : refuse [mdi:trash-can]
    2023-08-31 : refuse [mdi:trash-can]
    2023-09-07 : recycling [mdi:recycle]
    2023-09-07 : refuse [mdi:trash-can]
    2023-09-14 : refuse [mdi:trash-can]
    2023-09-21 : recycling [mdi:recycle]
    2023-09-21 : refuse [mdi:trash-can]
    2023-09-28 : refuse [mdi:trash-can]
    2023-10-05 : recycling [mdi:recycle]
    2023-10-05 : refuse [mdi:trash-can]
    2023-10-12 : refuse [mdi:trash-can]
    2023-10-19 : recycling [mdi:recycle]
    2023-10-19 : refuse [mdi:trash-can]
    2023-10-26 : refuse [mdi:trash-can]
    2023-11-02 : recycling [mdi:recycle]
    2023-11-02 : refuse [mdi:trash-can]
    2023-11-09 : refuse [mdi:trash-can]
    2023-11-16 : recycling [mdi:recycle]
    2023-11-16 : refuse [mdi:trash-can]
    2023-11-23 : refuse [mdi:trash-can]
    2023-11-30 : recycling [mdi:recycle]
    2023-11-30 : refuse [mdi:trash-can]
    2023-12-07 : refuse [mdi:trash-can]
    2023-12-14 : recycling [mdi:recycle]
    2023-12-14 : refuse [mdi:trash-can]
    2023-12-21 : refuse [mdi:trash-can]
    2023-12-28 : recycling [mdi:recycle]
    2023-12-28 : refuse [mdi:trash-can]
```

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/west_norfolk_gov_uk.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            waste = item.findAll("img")
+            for w in waste:
+                entries.append(
+                Collection(
+                    date = datetime.strptime(dt, "%A %d %B %Y").date(),
+                    t = w["alt"],
+                    icon = ICON_MAP.get(w["alt"].upper())
+                )
+            )
+        '''
```
Comment (5ila5) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/west_norfolk_gov_uk.py:75]: please remove not used / commented out code

Comment (dt215git) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/west_norfolk_gov_uk.py]: done

**Comment by 5ila5** [comment]
Tanks for your contribution

**Pull request merged**